gin/src/codegen.rs

```diff
@@ -28,6 +29,9 @@ pub struct CompileConfig {
     external_types: Vec<ExternalType>,
     proto_files: Vec<PathBuf>,
     well_known_types: bool,
+
+    #[cfg(feature = "tonic")]
+    generate_services: bool,
```

Comment:
This could be always existing and be set to false if not, no?

Reply:
In theory, yes, but if tonic isn't compiled, then this provides no extra functionality.

The only reason to keep the field there would just be because you don't want the attribute noise.

Reply:
I mean also the skip function makes no sense when this feature exists no?

Reply:
Or do I want tonic dep and no service generated?